src/mono/wasm/Wasm.Build.Tests/BrowserRunner.cs

```diff
@@ -88,8 +88,16 @@ internal class BrowserRunner : IAsyncDisposable
 
         if (runTask.IsCompleted)
         {
-            var res = await runTask;
-            res.EnsureSuccessful();
+            try
+            {
+                var res = await runTask;
+                res.EnsureSuccessful();
+            }
+            catch (Exception ex)
+            {
+                _testOutput.WriteLine($"Exception: {ex}");
+                throw;
```

Comment:
This one was not logged correctly, it's when I was getting
```
Exception: System.InvalidOperationException: No process is associated with this object.
     at System.Diagnostics.Process.EnsureState(State state)
     at System.Diagnostics.Process.get_HasExited()
     at Wasm.Build.Tests.ToolCommand.Dispose() in /workspaces/runtime/src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs:line 96
     at Wasm.Build.Tests.BlazorWasmTestBase.BlazorRunTest(String runArgs, String workingDirectory, BlazorRunOptions runOptions) in /workspaces/runtime/src/mono/wasm/Wasm.Build.Tests/Blazor/BlazorWasmTestBase.cs:line 230
     at Wasm.Build.Tests.TestAppScenarios.AppTestBase.RunSdkStyleApp(RunOptions options, BlazorRunHost host) in /workspaces/runtime/src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/AppTestBase.cs:line 119
```
Wrapping this line it in a try-catch revealed a trivial error:
```
An error occurred trying to start process '/workspaces/runtime/artifacts/bin/dotnet-latest/dotnet' with working directory '/workspaces/runtime/artifacts/bin/Wasm.Build.Tests/Debug/net9.0/linux-x64/wbt artifacts/SignalRClientTests_5z404rjp_ocr/App/bin/Debug/net9.0/publish/wwwroot'. No such file or directory
```
I decided to leave it for future time saving.

Reply:
I would like to understand why it wasn't logged correctly and solve the systemic issue. Otherwise we would have to put such try-catch everywhere I guess

Reply:
This line throws in the discussed case:
https://github.com/dotnet/runtime/blob/7b534da455fa12c7a19b518ddab059dd70af1fa7/src/mono/wasm/Wasm.Build.Tests/BrowserRunner.cs#L91

It's executed as a part of async task connected to `TaskCompletionSource` that we are waiting for in `RunSdkStyleApp`. If the `BlazorRunTest` action throws, we need to check `InnerException` property to see what were the details of the exception. By calling `ToString` on exception we collect: `Message`, `StackTrace`, and `InnerException`, that's why this fix helped me see what's wrong. To avoid putting try-catch everywhere, as you mentioned, I will move it out to `RunSdkStyleApp` where we create the `TaskCompletionSource`.